rust/lanzatool/src/generation.rs

```diff
@@ -1,42 +1,69 @@
+use serde::{Deserialize, Serialize};
+use serde::de::IntoDeserializer;
+
 use std::fmt;
 use std::fs;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 
-use anyhow::{Context, Result};
+use anyhow::{anyhow, Context, Result};
+use bootspec::generation::Generation as BootspecGeneration;
+use bootspec::BootJson;
+use bootspec::SpecialisationName;
 
-use crate::bootspec::Bootspec;
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SecureBootExtension {
+    #[serde(rename="osRelease")]
+    pub os_release: PathBuf
+}
 
 #[derive(Debug)]
 pub struct Generation {
+    /// Top-level nixpkgs version
     version: u64,
-    specialisation_name: Option<String>,
-    pub bootspec: Bootspec,
+    /// Top-level specialisation name
+    specialisation_name: Option<SpecialisationName>,
+    /// Top-level bootspec document
+    pub bootspec: BootJson,
```

Comment:
If we split up the bootspec from the extension I would really want to have both of these consolidated into a separate struct so we don't have to pass these values around together all the time.

Reply:
Can do that